miso-web/src/main/webapp/scripts/list_servicerecord.js

```diff
@@ -114,7 +115,17 @@ ListTarget.servicerecord = {
             list += "</ul>";
             return list;
           }
-          return data;
+          var filenames = "";
+          var counts = 0;
+          full.attachments.forEach(function (file) {
+            if (full.attachments.length - 1 == counts) {
+              filenames += file.filename;
+            } else {
+              filenames += file.filename + " ";
+              ++counts;
+            }
+          });
+          return filenames;
```

Comment:
This will error if `file.attachments` is null/undefined. Could restructure function like:

```
if (!file.attachments || !file.attachments.length) {
  return null;
} else if (type === "display") {
  ...
} else {
  ...
}
```

Reply:
Could there be a case where file attachments is null? From my understanding, when there is no file attachments, `file.attachments` will be an empty array and `file.attachments.length` would be zero. Hence, it would simply return an empty string without iterating through each file attachment.

Reply:
This check makes it more resilient to back-end changes, so there is less chance of future bugs

Reply:
I see. Will make a change now!